### ISSUE TITLE
gopass grep *pattern* should not fail if there are no matches

### ISSUE THREAD

**Issue by twpayne** [gopasspw/gopass#1426]
### Summary

`gopass grep ^$` where `^$` is a regexp that does not match any secrets causes gopass to exit with a failure exit code. A regexp not matching is not a failure.

### Steps To Reproduce

Run `gopass grep ^$`.

### Expected behavior

gopass exits with no output and a successful exit code (0), just like `grep`.

### Environment

- OS: macOS 10.15.5 (Catalina)
- OS version: Darwin Toms-MacBook-Pro.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
- gopass Version:
```
gopass 1.9.2 (2020-06-04 20:34:23) go1.14.3 darwin amd64
<root>     - gpg 2.2.20 - git 2.27.0 -   fs 0.1.0
Available Crypto Backends: age, gpgcli, plain, xc
Available RCS Backends: gitcli, noop
Available Storage Backends: fs, inmem
```
- Installation method: brew

### Additional context

Background info is at https://github.com/twpayne/chezmoi/issues/784
tl;dr: chezmoi needs a way to reliably cache the gpg passphrase used by pass/gopass and `gopass grep ^$` is a reasonable solution, except that gopass treats no matches as a failure.

**Comment by dominikschulz** [comment]
Wow, this sounds like a terrible hack. Decrypting all secrets in some of my password stores take up to an hour.
And that's just what `gopass grep ^$` is doing.

twpayne/chezmoi#784 is another good example why I eventually want to get rid of GPG.
It's awkward to interface with and has way too many possible deployment configurations.

Adhering to the grep exit code behavior sounds good however.

**Comment by twpayne** [comment]
Thank you for the speedy fix :)

**Comment by dominikschulz** [comment]
No problem.

But please don't complain is this takes a long time for some users.

Happy to discuss other options, but implementing any might take a lot longer.
